rust_src/src/process.rs

```diff
@@ -322,4 +327,19 @@ pub fn set_process_query_on_exit_flag(mut process: LispProcessRef, flag: LispObj
     flag
 }
 
+/// Return the exit status of PROCESS or the signal number that killed it.
+/// If PROCESS has not yet exited or died, return 0.
+#[lisp_fn]
+pub fn process_exit_status(process: LispProcessRef) -> LispObject {
+    if process.raw_status_new() {
+        unsafe { update_status(process.as_ptr()) };
+    }
+    let status = LispObject::from_raw(process.status);
+    if let Some(s) = status.as_cons() {
+        s.cdr().as_cons().unwrap().car()
```

Comment:
`car(s.cdr())` is what you want. You are correct in thinking this is not `XCAR`. But naked unwraps are not wise unless we are sure they will not fail.

One way to write this is:

    status.as_cons().map_or_else(|| LispObject::from(0),
                                 |cons| car(cons.cdr()))